### ISSUE TITLE
Partition bar on the assets view misses failed assets

### ISSUE THREAD

**Issue by aroig** [dagster-io/dagster#10161]
### Dagster version

commit ca87aa3c95204c7941e675318f714d24a4990132

### What's the issue?

On a repo with two assets, asset_1 and asset_2. Materialize both successfully. Then make asset_2 fail, and rematerialize them. 

After that, the partition bar on the assets view, or on the Materialize dialog, does not show asset_2 as failed, but as green. This is inconsistent with the partition bar on the Job > Partitions page, which shows as red.

I attach a picture of the materialize dialog, and the Partitions page for comparison.

![partitions](https://user-images.githubusercontent.com/1743143/197407557-d97b5f4d-5a0b-4fbc-b333-0b9336abbace.png)
![misleading-materialize](https://user-images.githubusercontent.com/1743143/197407560-763a0b4b-8ac1-48b3-b3ec-076611afc343.png)

### What did you expect to happen?

I woud expect to see assets with failed partitions marked as red on the partition bar for the asset, or on the Materialize dialog. Instead they misleadingly show as green.

### How to reproduce?

With the following code

``` python
import dagster

partitions = dagster.StaticPartitionsDefinition(["A", "B", "C"])

@dagster.asset(
    partitions_def=partitions,
    group_name="all",
)
def asset_1():
    return 1

@dagster.asset(
    partitions_def=partitions,
    group_name="all",
)
def asset_2(asset_1: int):
    should_fail = Fail

    if should_fail:
        raise RuntimeError("Failed, as requested!")

    return asset_1 + 1

job_all = dagster.define_asset_job(
    name="all",
    selection=dagster.AssetSelection.groups("all")
)

@dagster.repository
def my_repository():
    return [
        asset_1,
        asset_2,
        job_all,
    ]
```

* On the all job > Overview > Materialize all assets and partitions. All should succeed, and all asset partition combinations should show green on the Partitions matrix.
* Edit the code to set `should_fail = True` on asset_2. Refresh the repo, and rematerialize everything again. asset_2 should fail for all 3 partitions.
* Observe that Partitions page correctly shows a red partition bar, the dots for asset_2 are marked as failed. On the other hand, on Overview, click on asset_2, and observe that the partition bar on the bottom of the right pane is all green. Same if you open the Materialize dialog, asset_2 appears with a green bar.

### Deployment type

Local

### Message from the maintainers

Impacted by this issue? Give it a 👍! We factor engagement into prioritization.

**Comment by bengotow** [comment]
I think this is related to https://github.com/dagster-io/dagster/issues/10159 and we'll resolve it in upcoming releases, but the reasoning behind the difference is this:

- For op jobs, an "op x partition" box is green if the *last* run of that "op x partition"  succeeded 

- For asset jobs, an asset is green for a partition if it has been materialized for that partition

This means that subsequent failures turn the op x partition matrix red, while the asset version still sees that there's a valid materialization for that partition and remains green.

Note: We're adding "asset versioning" soon, which will bring a bit more nuance to the asset bars including the concept of a "stale" materialization. I think this could make the two look more similar in some cases.

**Comment by sryza** [comment]
This is addressed in Dagster 1.2, so going to close! Feel free to reopen if there's anything missing.
